misk-hibernate/src/main/kotlin/misk/hibernate/SchemaMigratorService.kt

```diff
@@ -22,8 +24,11 @@ class SchemaMigratorService internal constructor(
   override fun startUp() {
     val schemaMigrator = schemaMigratorProvider.get()
     if (environment == Environment.TESTING || environment == Environment.DEVELOPMENT) {
-      val appliedMigrations = schemaMigrator.initialize()
-      schemaMigrator.applyAll("SchemaMigratorService", appliedMigrations)
+      if (config.type != DataSourceType.VITESS) {
+        // vttestserver automatically applies migrations
```

Comment:
are there particular reasons why we shouldn't set it up instead of `vttestserver`? Does `vttestserver` do other special setup that existing tables would mess with? I ask because it would be nice to exercise applyAll for tests for vitess-based apps

Reply:
Yeah doing it in vttestserver is faster and easier because it runs it in parallel and it runs it before applying the vschema. It would be a lot of hard and quite slow to do it outside of vttestserver because we would have to call in to do all the set up. This is the main reason Vitess tests are so slow in the Square monorepo.

I don't think there's a point in exercising `applyAll` in Vitess tests. It's never used in production.